src/display/canvas.js

```diff
@@ -961,20 +985,23 @@ class CanvasGraphics {
         return i;
       }
 
-      fnId = fnArray[i];
+      if (!filter || filter(i, this[fnArray[i]]?.name)) {
```

Comment:
(note to myself) This is unrelated, I need to remove it from this PR. It's part of the logic that actually allows partial rendering using the data collected in this PR.